x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -243,22 +244,21 @@ export function systemRoutes(
         tags: ['access:ml:canAccessML'],
       },
     },
-    mlLicense.basicLicenseAPIGuard(async ({ legacyClient, request, response }) => {
+    mlLicense.basicLicenseAPIGuard(async ({ client, request, response }) => {
       try {
         const { index } = request.body;
 
         const options = {
           index: [index],
           fields: ['*'],
-          ignoreUnavailable: true,
-          allowNoIndices: true,
-          ignore: 404,
+          ignore_unavailable: true,
+          allow_no_indices: true,
```

Comment:
what about `ignore: 404` ? shall we add something to the `catch` section instead?

Reply:
`ignore` isn't allowed as an argument. it's not in the docs either. i think the legacy client silently ignored it.